arch/xtensa/core/cpu_idle.c

```diff
@@ -3,8 +3,15 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <zephyr/toolchain.h>
 #include <zephyr/tracing/tracing.h>
 
+#if defined(__XT_CLANG__) && defined(CONFIG_XTENSA_WAITI_BUG)
```

Comment:
Might as well make the optimization flag unconditional (maybe with a comment that xt-clang miscompiles it).  There's no value in compiler optimization to a function that literally does nothing anyway.

Reply:
Removed.